spacewalk/uyuni-setup-reportdb/bin/uyuni-setup-reportdb-user

```diff
@@ -0,0 +1,189 @@
+    DEFAULT=${CURRENT_VALUE:-$DEFAULT}
+    local VARIABLE_ISSET=$(set | grep "^$VARIABLE=")
+
+    echo -n "$MSG [$DEFAULT]: "
+    if [ "$INTERACTIVE" = "1" ] &&  [ -z "$VARIABLE_ISSET" ]; then
+        if [ "$SILENT" = 1 ]; then
+          read -sr INPUT
+        else
+          read -r INPUT
+        fi
+    elif [ -z "$VARIABLE_ISSET" ]; then
+        echo "$DEFAULT"
+    else
+        DEFAULT=${!VARIABLE}
+        echo "$DEFAULT"
+    fi
+    if [ -z "$INPUT" ]; then
+        if [ "$DEFAULT" = "y/N" ] || [ "$DEFAULT" = "Y/n" ]; then
+            INPUT=$(yes_no "$DEFAULT")
+        else
+            INPUT="$DEFAULT"
+        fi
+    fi
+    ACCUMULATED_ANSWERS+=$(printf "\n%q=%q" "$VARIABLE" "${INPUT:-$DEFAULT}")
+    eval "$(printf "%q=%q" "$VARIABLE" "$INPUT")"
+}
+
+set_value() {
+    local OPTION="$1"
+    local VAR="$2"
+    local ARG="$3"
+    [[ "$ARG" =~ ^- ]] \
+        && echo "$0: option $OPTION requires argument! Use answer file if your argument starts with '-'." \
+        && print_help
+    eval "$(printf "%q=%q" "$VAR" "$ARG")"
+}
+
+parse_properties() {
+    ATTRIBUTE="$1"
+    VAR="$2"
+    VALUE=$(grep "^$ATTRIBUTE" $RHN_CONF |cut -d'=' -f2 | tail -n1 | tr -d '\n ')
+    eval "$(printf "%q=%q" "$VAR" "$VALUE")"
+}
+
+parse_properties $DBNAME_ATTRIBUTE DBNAME
+
+if [ -z $DBNAME ]; then
+  echo "$DBNAME_ATTRIBUTE is missing. Exit"
+  exit 1
+fi
+
+OPTS=$(getopt --longoptions=help,non-interactive,add,delete,modify,action:,dbuser:,dbpassword: -n '${0##*/}' -- h "$@")
+
+if [ $? != 0 ] ; then
+    print_help
+fi
+
+# It is getopt's responsibility to make this safe
+eval set -- "$OPTS"
+
+ACTION_COUNT=0
+while : ; do
+    case "$1" in
+        --help|-h)  print_help;;
+        --non-interactive) INTERACTIVE=0;;
+        --dbuser) set_value "$1" DBUSER "$2"; shift;;
+        --dbpassword) set_value "$1" DBPASSWORD "$2"; shift;;
+        --add) ACTION="a" ACTION_COUNT=$((ACTION_COUNT+1)) ;;
+        --delete) ACTION="d" ACTION_COUNT=$((ACTION_COUNT+1)) ;;
+        --modify) ACTION="m" ACTION_COUNT=$((ACTION_COUNT+1)) ;;
+        --) shift;
+            if [ $# -gt 0 ] ; then
+                echo "Error: Extra arguments found: $@"
+                print_help
+                exit 1
+            fi
+            break;;
+        *) echo Error: Invalid option $1; exit 1;;
+    esac
+    shift
+done
+
+if [ "$INTERACTIVE" = "1" ] ; then
+  echo "Report DB Name is: $DBNAME"
+  for i in $(seq $INTERACTIVE_RETRIES) ; do
+    default_or_input "[a]dd/[m]odify/[d]elete user. Default is " ACTION 'm'
+      case "$ACTION" in
+        a|m|d)
+        break
+        ;;
+        *)
+          echo "Invalid input. Please [a]dd/[m]odify/[d]elete"
+          unset ACTION
+        ;;
+      esac
+  done
+
+  if [ -z $ACTION ]; then
+    echo "Invalid input. Exit"
+    exit 1
+  fi
+
+  default_or_input "User:" DBUSER ''
+
+  if [ "$ACTION" != "d" ]; then
+    default_or_input "Password:" DBPASSWORD '' '1'
+    echo
```

Comment:
I would really ask to Verify Password when you create the user. Only in interactive mode.